test/middleware.test.ts

```diff
@@ -558,4 +558,19 @@ describe("test/body-parser.test.ts", () => {
         .expect({ foo: "bar" });
     });
   });
+
+  describe("request closed", () => {
+    it("should return 499 on request closed", async () => {
+      const app = new Koa();
+
+      app.use(async (ctx, next) => {
+        Object.defineProperty(ctx.req, "closed", { value: true });
```

Comment:
maybe call `req.destroy()` is better than mock `closed` property?

https://nodejs.org/docs/latest/api/stream.html#readabledestroyerror

Reply:
I have tried multiple things to simulate a client closing a request for example `req.destroy()` and `req.emit("close")` ect, but they are all handled differently due to the server closing the request instead of the client.